In [10]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import ttest_ind

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


α == 0.05 (95% Confidence)

Y == Continious  
X == Discrete

Is Y1 and Y2 normal ?

H0 = Y1 and Y2 are same

H1 = Y1 and Y2 are not same 


## Normality Test

In [3]:
df = pd.read_csv("Cutlets.csv")

In [7]:
df.shape

(35, 2)

In [8]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [9]:
df.mean()

Unit A    7.019091
Unit B    6.964297
dtype: float64

In [5]:
stats.shapiro(df["Unit A"])

ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)

P value for Unit A == 0.32 > α

In [6]:
stats.shapiro(df["Unit B"])

ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)

(0.9727300405502319, 0.5224985480308533)  
P value for Unit B == 0.52 > α

### HO is accepted. Thats is both Y1 and Y2 are same

##  External Condition

* External condition are same.  
* Thus we can perform Paired T Test

##  Model - Paired t test

H0 == Mean for Y1 and Y2 are equal (There is no significance difference between diameter of the Culets)  
H1 == Mean for Y1 and Y2 are not equal (There is a significance difference between diameter of the Culets)

In [4]:
stats.ttest_rel(df["Unit A"], df["Unit B"])

Ttest_relResult(statistic=0.7536787225614314, pvalue=0.4562300768038412)

Ttest_relResult(statistic=0.7536787225614323, pvalue=0.45623007680384076)  
P value of the Paired T Test is == 0.45 > α  

## Thus H0 is accepted.  
## Mean of both Y1 and Y2 is equal